### PR TITLE
torch.compile for faster epochs (more iterations in 5min)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
We're bottlenecked at ~10 epochs in 5 minutes (~31s/epoch). torch.compile can fuse operations and optimize the computation graph, potentially reducing epoch time by 20-40%. Even 2-3 extra epochs could meaningfully improve results since the model is still converging at epoch 10.

**Risk:** Compilation overhead (first epoch is slow), and dynamic shapes from variable-size meshes may prevent some optimizations.

## Instructions

In `train.py`, after model creation (line 79), add torch.compile:

```python
model = Transolver(
    **model_config
).to(device)

# Add this line:
model = torch.compile(model)
```

**Note:** If this crashes with dynamic shape errors, try `torch.compile(model, dynamic=True)` instead.

**No other changes.** Run:
```bash
python train.py --agent fern --wandb_group mar14b-compile --wandb_name "fern/compile"
```

## Baseline
| Metric | Value |
|--------|-------|
| **surf_p** | **107.35** |
| surf_Ux | 1.23 |
| surf_Uy | 0.84 |
| epochs | ~10 in 5min |

---

## Results

**W&B run**: `17ymd0pi`
**Epochs completed**: 10 (wall-clock 5.2 min)
**Best epoch**: 10
**Peak VRAM**: ~15.4 GB
**Epoch time**: 24.2s (vs 31s baseline, **-22%**)

| Metric | torch.compile | Baseline | Delta |
|---|---|---|---|
| val/loss | 2.5536 | 2.45 | +0.10 |
| Surface MAE Ux | 1.15 | 1.23 | -0.08 |
| Surface MAE Uy | 0.84 | 0.84 | 0 |
| Surface MAE p | 113.09 | **107.35** | **+5.74 (+5.3%)** |
| Volume MAE Ux | 5.57 | 5.44 | +0.13 |
| Volume MAE Uy | 2.61 | 2.57 | +0.04 |
| Volume MAE p | 144.8 | 152.5 | -7.7 |
| Epoch time | **24.2s** | 31.0s | **-22%** |

**What happened**: torch.compile succeeded and reduced per-epoch time by 22% (31s → 24.2s). However, the **compilation overhead** (~80s for first epoch) consumed the time saved — we still only completed 10 epochs in 5.2 minutes. No extra epochs were gained. surf_p regressed slightly (107.35 → 113.09). The hypothesis was sound but the implementation risk materialized: compilation overhead cancels the speedup for short training runs. Vol_p improved notably (152.5 → 144.8).

**Suggested follow-ups**:
- If the training budget is extended (e.g., 10+ min), torch.compile would provide genuine benefit.
- Within the current 5-min budget, compile adds more harm (compilation overhead) than good.
- This negative result is about budget fit, not about compile itself.

### CODE REVIEW

**Comment by tcapelle** [comment]
Closing — torch.compile reduced epoch time 22% but compilation overhead (80s) ate the savings. Still only 10 epochs, and surf_p regressed to 113.09. Good finding though — compile would help with longer budgets.
